chromium_src/chrome/browser/ui/startup/infobar_utils.cc

```diff
@@ -33,5 +35,35 @@ class BraveGoogleKeysInfoBarDelegate {
 #undef ObsoleteSystemInfoBarDelegate
 #endif
 
+#undef AddInfoBarsIfNecessary
 #undef GoogleApiKeysInfoBarDelegate
 #undef ShowIfNotOffTheRecordProfile
+
+void AddInfoBarsIfNecessary(Browser* browser,
+                            Profile* profile,
+                            const base::CommandLine& startup_command_line,
+                            chrome::startup::IsFirstRun is_first_run,
+                            bool is_web_app) {
+  AddInfoBarsIfNecessary_ChromiumImpl(browser, profile, startup_command_line,
+                                      is_first_run, is_web_app);
+
+  if (IsKioskModeEnabled()) {
+    return;
+  }
+
+  if (!startup_command_line.HasSwitch(switches::kTestType) &&
+      !IsAutomationEnabled()) {
+    static bool infobars_shown = false;
+    if (infobars_shown) {
+      return;
+    }
+    infobars_shown = true;
+
+    content::WebContents* web_contents =
+        browser->tab_strip_model()->GetActiveWebContents();
+    DCHECK(web_contents);
```

Comment:
is this really true? I see this check in the ChromiumImpl:
```
  if (!browser || !profile || browser->tab_strip_model()->count() == 0)
    return;
```

Reply:
Yeah, added this checking. thanks 👍🏼